### ISSUE TITLE
`$elemMatch` operator does not work in the new SDK

### ISSUE THREAD

**Issue by kaceycleveland** [BuilderIO/builder#3007]
**Describe the bug**

We are in the process of porting over to the new SDK (which I saw was marked as 1.0.0, nice!)
Unfortunately, the old functionality of the API could not be replicated with the new API with `$elemMatch`.

**To Reproduce**
Not a concrete reproduction but my test script is as follows:
```ts
// @ts-nocheck ignore all issues as this is a script
import { builder } from '@builder.io/sdk';
import { fetchEntries } from '@builder.io/sdk-react';

builder.init(process.env.NX_BUILDER_IO_PUBLIC_KEY);

const callBuilder = async () => {
  const results = await builder.getAll('blog-article', {
    options: {
      sort: {
        createdDate: -1,
      },
      enrich: true,
    },
    query: {
      'data.categories': {
        $elemMatch: {
          'category.id': {
            $in: JSON.stringify(['d491820ec5da422199336a73f388f354']).trim(),
          },
        },
      },
    },
    limit: 5,
    omit: 'data.blocks',
  });
  const resultsNew = await fetchEntries({
    model: 'blog-article',
    apiKey: process.env.NX_BUILDER_IO_PUBLIC_KEY!,
    options: {
      sort: {
        createdDate: -1,
      },
      enrich: true,
    },
    query: {
      'data.categories': {
        $elemMatch: {
          'category.id': {
            $in: JSON.stringify(['d491820ec5da422199336a73f388f354']).trim(),
          },
        },
      },
    },
    limit: 5,
    omit: 'data.blocks',
  });

  const pages = results.map((res) => res.data.title);
  pages.forEach((article) => console.log(article));

  if (results.length && results[0]) {
    console.log(Object.keys(results[0].data));
    console.log(results[0].data.categories);
  }

  console.log('NEW -------------');

  const pagesNew = resultsNew.map((res) => res.data.title);
  pagesNew.forEach((article) => console.log(article));

  if (resultsNew.length && resultsNew[0]) {
    console.log(Object.keys(resultsNew[0].data));
    console.log(resultsNew[0].data.categories);
  }
};

callBuilder();

```

**Expected behavior**
The return in the repo test script should be the same results but the new API returns no results.
It would be helpful to have a debug option attached to the builder API client to output some additional information such as the query keys being passed to the request or something.

**Screenshots**

**Additional context**

**Comment by kaceycleveland** [comment]
Update from the above post, the work around and the example query provided in the above thread **do not work**.

From my POV, the `$elemMatch` operator seems to be broken or at least I have yet to be able to find a query that returns the correct results.

Mentioned in the thread, but I really feel like this is an important functionality to cover especially when we were told we should be using the new SDK. I would take a stab at creating a PR to address this in the new SDK but I would first need to find out if its even a problem with the SDK itself or if its the new content API. So far, I have not been able to get it working properly with the API query itself so that would be the first step if someone could provide a working example.

**Comment by chub7** [comment]
We encountered the same issue with the $elemMatch. Looking forward to a resolution.

**Comment by mikehuebner** [comment]
Hey y'all, any updates on this?

**Comment by samijaber** [comment]
A fixed was deployed in 1.0.27 https://github.com/BuilderIO/builder/releases/tag/%40builder.io%2Fsdk-react%401.0.27
